src/test/lib/ApiTest.cpp

```diff
@@ -579,6 +579,95 @@ void QuicTestValidateConnection()
     // ConnectionClose null handle.
     //
     MsQuic->ConnectionClose(nullptr);
+
+    //
+    // Invalid datagram send calls.
+    //
+    {
+        ConnectionScope Connection;
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->ConnectionOpen(
+                Session,
+                DummyConnectionCallback,
+                nullptr,
+                &Connection.Handle));
+
+        uint8_t RawBuffer[] = "datagram";
+        QUIC_BUFFER DatagramBuffer = { sizeof(RawBuffer), RawBuffer };
+
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->DatagramSend(
+                Connection.Handle,
+                nullptr,
+                1,
+                QUIC_SEND_FLAG_NONE,
+                nullptr));
+
+        TEST_QUIC_STATUS(
+            QUIC_STATUS_INVALID_PARAMETER,
+            MsQuic->DatagramSend(
+                Connection.Handle,
+                &DatagramBuffer,
+                0,
+                QUIC_SEND_FLAG_NONE,
+                nullptr));
+    }
+
+    //
+    // Successful send datagram calls.
+    //
+    {
+        ConnectionScope Connection;
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->ConnectionOpen(
+                Session,
+                DummyConnectionCallback,
+                nullptr,
+                &Connection.Handle));
+
+        uint8_t RawBuffer[] = "datagram";
+        QUIC_BUFFER DatagramBuffer = { sizeof(RawBuffer), RawBuffer };
+
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->DatagramSend(
+                Connection.Handle,
+                &DatagramBuffer,
+                1,
+                QUIC_SEND_FLAG_NONE,
+                nullptr));
+    }
+
+    //
+    // Successful set datagram receive parameter.
+    //
+    {
+        ConnectionScope Connection;
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->ConnectionOpen(
+                Session,
+                DummyConnectionCallback,
+                nullptr,
+                &Connection.Handle));
+
+        BOOLEAN ReceiveDatagrams = TRUE;
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->SetParam(
+                Connection.Handle,
+                QUIC_PARAM_LEVEL_CONNECTION,
+                QUIC_PARAM_CONN_DATAGRAM_RECEIVE_ENABLED,
+                sizeof(ReceiveDatagrams),
+                &ReceiveDatagrams));

```

Comment:
You should add a GetParam call here to verify the setting changed appropriately.

Reply:
I think the DatagramNegotiate test covers that area enough already. I don't really care to push another change just for the Get case.